src/ComputationContainer/Share/Share.hpp

```diff
@@ -470,9 +470,18 @@ auto _sqrt(const T &x)
 template <typename T, typename BT = float>
 Share<T> getRandBitShare()
 {
+    Config *conf = Config::getInstance();
     while (true)
     {
-        auto r = Share<BT>(RandGenerator::getInstance()->getRand<BT>(-10, 10));
+        auto r_int = RandGenerator::getInstance()->getRand<int>(-10, 10);
+        // SPは奇数，それ以外は偶数になるまで再生成
+        while ((conf->party_id == conf->sp_id && !(r_int & 1))
+               || (conf->party_id != conf->sp_id && (r_int & 1)))
+        {
+            r_int = RandGenerator::getInstance()->getRand<int>(-10, 10);
+        }
```

Comment:
If we take the method of creating odd and even numbers by arithmetic, we don't need to regenerate random numbers.

```suggestion
        // this suggestion is not tested
        auto r_int = RandGenerator::getInstance()->getRand<int>(-5, 5) * 2;  // generate even number
        if (conf->party_id == conf->sp_id)
        {
            ++r_int;  // note: this range number range is `[-9, 11]`
        }
```

On the other hand, the cost is small compared to others, so I think it's okay to leave it as it is.
@mdonaka 
What do you think?

Reply:
Thanks for the suggestion! This one is much more beautiful!
No cost to fix, so I'll see if it works and fix it.

Reply:
@arukuka 
Update with 31bd3302a519dd5e5b50fa9c4db8c5d354523904! Could you check it?

Reply:
@mdonaka 
Cool!  Thank you!